### PR TITLE
Generate static HTML of the manual to serve on Github Pages

### PR DESCRIPTION
**Pull request by aaanh**
This PR addresses issue #19 
- Fixed one particular `file` command in `speed-manual.tex` that escapes during the HTML rendering.
- Created configurations for `make4ht`, the tool to generate HTML and CSS from `speed-manual.tex`.
    - Ignored artifacts created by the `make4ht` tool.
    - Generated HTML, CSS outputs in [doc/web](doc/web).
    - Added html build steps in [doc/Makefile](doc/Makefile).
- Added [doc/.devcontainer](doc/.devcontainer).
- Added [.github](.github) to trigger static page deployment to Github Pages.

### CODE REVIEW

**Review by smokhov — CHANGES_REQUESTED** [review]
Unfortunately this is a wrong way to fix. The \ there is for a reason. We need to find a post-processing step to strip them instead. Right now the LaTeX manual does not compile properly it produces 20 errors...

File: doc/.devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      # For Windows without WSL, a copy will be made
+      # from /tmp/.ssh to ~/.ssh to fix permissions
+      # - ~/.ssh:/tmp/.ssh:ro
+      # Shell history persistence
+      - ~/.zsh_history:/root/.zsh_history:z
+      # Git config
+      - ~/.gitconfig:/root/.gitconfig
+    environment:
+      - TZ=
+    entrypoint: ["zsh", "-c", "while sleep 1000; do :; done"]
```
Comment (smokhov) [inline doc/.devcontainer/docker-compose.yml:22]: Can you explain why the while loop like this is in the entrypoint?

Comment (aaanh) [reply doc/.devcontainer/docker-compose.yml:22]: This is to work around a bug from VS Code >= 1.64 (currently v1.73) when spinning up the containers using the extension. The work around was detailed in this issue on the VS Code's repository: https://github.com/microsoft/vscode-remote-release/issues/6337. From my understanding, it's a race condition that prevents the dev container from being gracefully started and VS Code to remote into the container.

**Comment by aaanh** [comment]
I have reverted the changes in the source `doc/speed-manual.tex` and tested out the PDF build to be successful.

File: doc/.devcontainer/docker-compose.yml (hunk 1 above)
Comment (smokhov) [reply doc/.devcontainer/docker-compose.yml:22]: Thanks. It'd be good to add a comment to that effect in this yml file for those who are not in the know.

Comment (aaanh) [reply doc/.devcontainer/docker-compose.yml:22]: Got it, I will add more details in the `.devcontainer/README.md` on my next push

**Comment by aaanh** [comment]
With the latest push:
- `.devcontainer/` has been moved to `src/.devcontainer`
-  Changed container volume mounting to include the project root in a nested workspace
- Updated devcontainer README with more details on the configuration files

**Comment by smokhov** [comment]
Thanks @aaanh for the changes so far. I don't see `doc/Makefile` changes though in this PR... Like where are the steps to build the HTML manual. And where it should be running, in the container?

**Comment by smokhov** [comment]
Thanks for the contribution!

**Pull request merged**